src/rest_api/routing.clj

```diff
@@ -43,7 +43,7 @@
           entity-class (-> (:context request)
                            (str/split #"/")
                            (last))
-          attr (keyword entity-class "id")
+          attr (keyword (str/replace entity-class #"_" "-") "id")
```

Comment:
I think it'd be preferable to pass in a `entity-class` to the `make-request-handler` function as a new parameter; either as is, or just pass the `route-data` mapping, and get `entity-class` from that `(:entity-class route-data)`).